src/components/PersonalPage.vue

```diff
@@ -0,0 +1,179 @@
+export default {
+    name: 'Personal-Page',
+    data: () => ({
+        addictions: [],
+        addiction: "",
+        promises: [],
+        promise: "",
+        reasons: [],
+        reasonDescription: "",
+        name: '',
+        description: '',
+        days: "",
+        dialog: false,
+        switchAddictionPromises: false,
+    }),
+    methods: {
+        async loadAddictions() {
+            const answer = axios.get(`http://localhost:8080/addiction/all`);
+            answer.then(body => {
+                this.addictions = body.data;
+            }
+            )
+        },
+        async loadPromises() {
+            const answer = axios.get(`http://localhost:8080/user/getPromises?userName=${this.$route.params.userName}`);
+            answer.then(body => {
+                this.promises = body.data;
+            }
+            )
+        },
+        async addNewAddiction() {
+            const answer = axios.post(`http://localhost:8080/addiction/add`, { name: this.name, description: this.description });
+            answer.then(body => {
+                this.addictions = body.data;
+            }
+            )
+            this.name = '';
+            this.description = '';
+        },
+        async addNewPromise() {
+            const answer = axios.post(`http://localhost:8080/user/addPromise?userName=${this.$route.params.userName}&addiction_id=${this.addiction.addictionId}&days=${this.days}`);
+            answer.then(body => {
+                this.promises = body.data;
+            }
+            )
+            this.addiction = '';
+            this.days = '';
+        },
+        async deletePromise(promise) {
+            const answer = axios.post(`http://localhost:8080/promise/deletePromise?userName=${this.$route.params.userName}&promiseId=${promise.promiseId}`);
+            answer.then(body => {
+                this.promises = body.data;
+            }
+            )
+            this.addiction = '';
+            this.days = '';
+        },
+        async loadReasons(promise) {
+            const answer = axios.get(`http://localhost:8080/reason/getReasons?promiseId=${promise.promiseId}`);
+            answer.then(body => {
+                this.reasons = body.data;
+            }
+            )
+            this.promise = promise;
+            console.log(this.promise)
+            this.dialog = true
+        },
+
+        async addNewReason() {
+            const answer = axios.post(`http://localhost:8080/reason/addReason?promiseId=${this.promise.promiseId}&description=${this.reasonDescription}`);
+            answer.then(body => {
+                this.reasons = body.data;
+            }
+            )
+            this.reasonDescription = '';
+        },
+
+        closeReasonsDialog() {
+            this.promise = "";
+            this.reasons = [];
+            this.dialog = false;
+        }
+    },
+    mounted() {
+        this.loadAddictions();
+        this.loadPromises();
+    }
+}
+</script>
```

Comment:
The component looks very overloaded. Think about splitting it into multiple components